tests/module_api/test_api.py

```diff
@@ -635,6 +637,30 @@ def test_check_push_rules_actions(self) -> None:
             [{"set_tweak": "sound", "value": "default"}]
         )
 
+    def test_create_room(self) -> None:
+        """Test that modules can create a room."""
+        # First test user existence verification.
+        self.get_failure(
+            self.module_api.create_room(
+                user_id="@user:test", config={}, ratelimit=False
+            ),
+            SynapseError,
+        )
+
+        # Now do the happy path.
+        user_id = self.register_user("user", "password")
+
+        (result, _) = self.get_success(
+            self.module_api.create_room(user_id=user_id, config={}, ratelimit=False)
+        )
+        room_id = result["room_id"]
+
+        channel = self.make_request(
+            "GET",
+            f"/_matrix/client/r0/directory/list/room/{room_id}",
```

Comment:
I think trying to fetch the `m.room.create` event in the room's state (`GET /_matrix/client/v3/rooms/[room ID]/state/m.room.create`) and comparing the sender with the room's creator is a better indication of success.

Reply:
Changed here: 62fc9c8